packages/react-dom-bindings/src/client/ReactFiberConfigDOM.js

```diff
@@ -2331,6 +2332,23 @@ function removeEventListenerFromChild(
   return false;
 }
 // $FlowFixMe[prop-missing]
+FragmentInstance.prototype.dispatchEvent = function (
+  this: FragmentInstanceType,
+  event: Event,
```

Comment:
We should check if this `event.bubbles` is true because if it's false, it should only fire on the fragment's own listeners and no the parent node. This can be solved by the same technique as self listeners.